### PR TITLE
更新hooks文件，新增pages文件-已评审

### PR DESCRIPTION
**Pull request by bgfNGl35QpGme9mpx0f4b8yE3S**
[#222](https://github.com/TestUpCommunity/TUGithubAPI/issues/222)
[#221](https://github.com/TestUpCommunity/TUGithubAPI/issues/221)
[#220](https://github.com/TestUpCommunity/TUGithubAPI/issues/220)
[#219](https://github.com/TestUpCommunity/TUGithubAPI/issues/219)
[#218](https://github.com/TestUpCommunity/TUGithubAPI/issues/218)
[#217](https://github.com/TestUpCommunity/TUGithubAPI/issues/217)
[#216](https://github.com/TestUpCommunity/TUGithubAPI/issues/216)
[#256](https://github.com/TestUpCommunity/TUGithubAPI/issues/256)

[#128](https://github.com/TestUpCommunity/TUGithubAPI/issues/128)
[#129](https://github.com/TestUpCommunity/TUGithubAPI/issues/129)
[#130](https://github.com/TestUpCommunity/TUGithubAPI/issues/130)
[#132](https://github.com/TestUpCommunity/TUGithubAPI/issues/132)
[#133](https://github.com/TestUpCommunity/TUGithubAPI/issues/133)
[#134](https://github.com/TestUpCommunity/TUGithubAPI/issues/134)

**/api/projects/projects.py**
[#411](https://github.com/TestUpCommunity/TUGithubAPI/issues/411)
[#410](https://github.com/TestUpCommunity/TUGithubAPI/issues/410)
[#409](https://github.com/TestUpCommunity/TUGithubAPI/issues/409)
[#408](https://github.com/TestUpCommunity/TUGithubAPI/issues/408)
[#407](https://github.com/TestUpCommunity/TUGithubAPI/issues/407)
[#406](https://github.com/TestUpCommunity/TUGithubAPI/issues/406)
[#405](https://github.com/TestUpCommunity/TUGithubAPI/issues/405)
[#404](https://github.com/TestUpCommunity/TUGithubAPI/issues/404)
[#403](https://github.com/TestUpCommunity/TUGithubAPI/issues/403)

**/api/projects/columns.py**
[#427](https://github.com/TestUpCommunity/TUGithubAPI/issues/427)
[#426](https://github.com/TestUpCommunity/TUGithubAPI/issues/426)
[#425](https://github.com/TestUpCommunity/TUGithubAPI/issues/425)
[#424](https://github.com/TestUpCommunity/TUGithubAPI/issues/424)
[#423](https://github.com/TestUpCommunity/TUGithubAPI/issues/423)
[#422](https://github.com/TestUpCommunity/TUGithubAPI/issues/422)

**/api/projects/cards.py**
#417 
#416 
#415 
#414 
#413 
#412 

**/api/projects/collaborators.py**
#421 
#420 
#419 
#418

### CODE REVIEW

File: api/repositories/pages.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    def request_a_page_build(self, owner, repo, access = False):
+        """
+        https://developer.github.com/v3/repos/pages/#request-a-page-build
+        """
+        if access:
+            headers= {'Accept':'application/vnd.github.switcheroo-preview+json'}
+            return self.post("/repos/{}/{}/pages/builds".format(owner, repo), headers)
+        return self.post("/repos/{}/{}/pages/builds".format(owner, repo))
+
+    def update_information_about_a_pages_site(self, owner, repo, developers = False, **kwargs):
```
Comment (zhangting85) [inline api/repositories/pages.py]: 文档中说：
Note: The GitHub Pages API contains two additional fields in responses, which developers can preview: html_url and source. To see these two new fields, you must provide a custom media type in the Accept header:

意思是如果我们在发请求的时候header带上这个指定的media type，那么在响应中会多两个字段。

因此我们这里不需要做是否是developers（开发者）的判断，我们调用这个接口的人当然是开发者了。

File: api/repositories/pages.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        """
+        https://developer.github.com/v3/repos/pages/#update-information-about-a-pages-site
+        """
+        if developers:
+            headers = {'Accept':'application/vnd.github.mister-fantastic-preview+json'}
+            return  self.put("/repos/{}/{}/pages".format(owner, repo), headers = headers, **kwargs)
```
Comment (zhangting85) [inline api/repositories/pages.py]: 在这里用headers = headers是不对的。
=是赋值符号，意思是直接用你写的headers替换掉原有的headers，但显然你这里不应该把原来headers里的用户名、token等信息覆盖掉。这里，应该把header视为一个dict，然后覆盖这个dict里的Accept字段。

也就是这样：
from copy import deepcopy #这句话放到整个文件的最上面
copied_headers=deepcopy(self.session.headers） #获取原来的headers，注意deep copy保证这是两个对象。
copied_headers['Accept']=xxxx #修改这个复制出来的header的accept字段
return self.put("/repos/{}/{}/pages".format(owner, repo), headers = copied_headers, **kwargs) 
#用复制并修改过的header发请求，因为是deepcopy，也不会影响其他请求里的header。

File: api/repositories/pages.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        """
+        https://developer.github.com/v3/repos/pages/#list-pages-builds
+        """
+        return self.get("/repos/{}/{}/pages/builds".format(owner, repo))
+
+    def request_a_page_build(self, owner, repo, access = False):
+        """
+        https://developer.github.com/v3/repos/pages/#request-a-page-build
+        """
+        if access:
```
Comment (zhangting85) [inline api/repositories/pages.py]: Note: This endpoint is currently available for developers to preview. During the preview period, the API may change without advance notice. To access this endpoint during the preview period, you must provide a custom media type in the Accept header:

文档中的这句话意思是，如果你要用这个接口，就修改header里的accept字段。
所以这里不需要if else，默认就是要用，然后headers字段的修改方法见我的另一个评论。

File: api/repositories/pages.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+            headers = {'Accept': 'application/vnd.github.mister-fantastic-preview+json'}
+            return self.post("/repos/{}/{}/pages".format(owner, repo), headers = headers, **kwargs)
+        else:
+            return self.post("/repos/{}/{}/pages".format(owner, repo), **kwargs)
+
+    def get_information_about_a_pages_site(self, owner, repo, developers = False):
+        """
+        https://developer.github.com/v3/repos/pages/#get-information-about-a-pages-site
+        """
+        if developers:
```
Comment (zhangting85) [inline api/repositories/pages.py]: 同上。

File: api/repositories/pages.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            headers = {'Accept':'application/vnd.github.switcheroo-preview+json'}
+            self.delete("/repos/{}/{}/pages".format(owner, repo),headers = headers)
+        else:
+            self.delete("/repos/{}/{}/pages".format(owner, repo))
+
+    def enable_a_pages_site(self, owner, repo, access  = False, developers = False, **kwargs):
+        """
+        https://developer.github.com/v3/repos/pages/#enable-a-pages-site
+        """
+        if access:
```
Comment (zhangting85) [inline api/repositories/pages.py]: 同上，不需要条件判断。这里，accept字段里应该是要同时加入两个值。

File: api/repositories/pages.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+            headers = {'Accept':'application/vnd.github.mister-fantastic-preview+json'}
+            return  self.put("/repos/{}/{}/pages".format(owner, repo), headers = headers, **kwargs)
+        else:
+            return  self.put("/repos/{}/{}/pages".format(owner, repo), **kwargs)
+
+    def disable_a_pages_site(self, owner, repo, developers = False):
+        """
+        https://developer.github.com/v3/repos/pages/#disable-a-pages-site
+        """
+        if developers:
```
Comment (zhangting85) [inline api/repositories/pages.py]: 同上

File: api/repositories/pages.py (hunk 2 above)
Comment (bgfNGl35QpGme9mpx0f4b8yE3S) [reply api/repositories/pages.py]: 感谢挺哥指正

**Review by zhangting85 — APPROVED** [review]
通过

**Pull request merged**